File: core/src/main/java/com/views/menu/residemenu/ResideMenuItem.java
package com.views.menu.residemenu;

import android.content.Context;
import android.view.LayoutInflater;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.R;
import com.core.utilities.LUIUtil;
import com.utils.util.ConvertUtils;

public class ResideMenuItem extends LinearLayout {

    /**
     * menu item  icon
     */
    private ImageView ivIcon;
    /**
     * menu item  title
     */
    private TextView tvTitle;

    public ResideMenuItem(Context context) {
        super(context);
        initViews(context);
    }

    public ResideMenuItem(Context context, int icon, int title) {
        super(context);
        initViews(context);
        ivIcon.setImageResource(icon);
        tvTitle.setText(title);
    }

    public ResideMenuItem(Context context, int icon, String title) {
        super(context);
        initViews(context);
        ivIcon.setImageResource(icon);
        tvTitle.setText(title);
    }

    private void initViews(Context context) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.residemenu_item, this);
        ivIcon = findViewById(R.id.iv_icon);
        tvTitle = findViewById(R.id.tvTitle);
    }

    /**
     * set the icon color;
     *
     * @param icon
     */
    public void setIcon(int icon) {
        ivIcon.setImageResource(icon);
    }

    /**
     * set the title with resource
     * ;
     *
     * @param title
     */
    public void setTitle(int title) {
        tvTitle.setText(title);
    }

    /**
     * set the title with string;
     *
     * @param title
     */
    public void setTitle(String title) {
        tvTitle.setText(title);
    }

    public ImageView getIvIcon() {
        return ivIcon;
    }

    public TextView getTvTitle() {
        return tvTitle;
    }

    public void setTextColor(int color) {
        tvTitle.setTextColor(color);
    }

    public void setTextSize(int typedValue, int size) {
        LUIUtil.INSTANCE.setTextSize(tvTitle, typedValue, size);
    }

    public void setTextShadow(int color) {
        LUIUtil.INSTANCE.setTextShadow(tvTitle, color);
    }

    public void setIvIconSizePx(int sizeInPx) {
        ivIcon.getLayoutParams().width = sizeInPx;
        ivIcon.getLayoutParams().height = sizeInPx;
        ivIcon.requestLayout();
    }

    public void setIvIconSizeDp(int sizeInDp) {
        ivIcon.getLayoutParams().width = ConvertUtils.dp2px(sizeInDp);
        ivIcon.getLayoutParams().height = ConvertUtils.dp2px(sizeInDp);
        ivIcon.requestLayout();
    }
}

File: core/src/main/java/com/loitpcore/views/menu/resideMenu/ResideMenuItem.kt
package com.loitpcore.views.menu.resideMenu

import android.content.Context
import android.view.LayoutInflater
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.loitpcore.R
import com.loitpcore.core.utilities.LUIUtil.Companion.setTextShadow
import com.loitpcore.core.utilities.LUIUtil.Companion.setTextSize
import com.loitpcore.utils.util.ConvertUtils

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
class ResideMenuItem : LinearLayout {
    @Suppress("unused")
    var ivIcon: ImageView? = null
    var tvTitle: TextView? = null

    constructor(context: Context) : super(context) {
        initViews(context)
    }

    @Suppress("unused")
    constructor(context: Context, icon: Int, title: Int) : super(context) {
        initViews(context)

        ivIcon?.setImageResource(icon)
        tvTitle?.setText(title)
    }

    constructor(context: Context, icon: Int, title: String?) : super(context) {
        initViews(context)

        ivIcon?.setImageResource(icon)
        tvTitle?.text = title
    }

    private fun initViews(context: Context) {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        inflater.inflate(R.layout.residemenu_item, this)

        ivIcon = findViewById(R.id.ivIcon)
        tvTitle = findViewById(R.id.tvTitle)
    }

    /**
     * set the icon color;
     *
     */
    fun setIcon(icon: Int) {
        ivIcon?.setImageResource(icon)
    }

    /**
     * set the title with resource
     * ;
     *
     */
    fun setTitle(title: Int) {
        tvTitle?.setText(title)
    }

    /**
     * set the title with string;
     *
     */
    fun setTitle(title: String?) {
        tvTitle?.text = title
    }

    fun setTextColor(color: Int) {
        tvTitle?.setTextColor(color)
    }

    fun setTextSize(size: Float) {
        setTextSize(textView = tvTitle, size = size)
    }

    fun setTextShadow(color: Int) {
        setTextShadow(textView = tvTitle, color = color)
    }

    @Suppress("unused")
    fun setIvIconSizePx(sizeInPx: Int) {
        ivIcon?.apply {
            layoutParams.width = sizeInPx
            layoutParams.height = sizeInPx
            requestLayout()
        }
    }

    fun setIvIconSizeDp(sizeInDp: Int) {
        ivIcon?.apply {
            layoutParams.width = ConvertUtils.dp2px(sizeInDp.toFloat())
            layoutParams.height = ConvertUtils.dp2px(sizeInDp.toFloat())
            requestLayout()
        }
    }
}
